### PR TITLE
Make save as command not overwrite original file and open the new file.

### LINKED ISSUE #8373: "Save As..." should open the new file
### Bug Description:
Currently "Save as..." will keep the original file open, instead of opening the new file which is very unintuitive as most (all?) programs open the new file. In particular, vscode will open the new file. 

### Steps to Reproduce:

1. Execute "Save as..."

### Additional Information

- Operating System: Linux
- Theia Version: Current Gitpod version

### LINKED ISSUE #8504: "Save as..." should not save changes to original file
### Bug Description:
Currently "Save as..." will save any changes made to the original file to the original file, then copy it to the new file. This is contrary to how most programs handle it, where the changes made are only saved to the new file.

### Steps to Reproduce:

1. Open a file and make some changes
2. Execute "Save As..."

### Additional Information

- Operating System: macOS
- Theia Version: Current Gitpod version

### PR DESCRIPTION
**Pull request by leopoldwe**
Signed-off-by: Leopold Eckert <leopoldwe@gmail.com>

#### What it does

Fixes #8373 
Fixes #8504

When using the "save as" command, a new file will be created with the new changes, and the old file will be unchanged. The text editor will then close the old file and open the new one.

#### How to test

1. Open a file and make some changes
2. Use "Save as" command
3. The text editor will now open the new file created with the changes. The old file will be closed.
4. The old file will be unchanged.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/workspace/src/browser/workspace-frontend-contribution.ts [hunk 1]
```diff
@@ -417,12 +417,18 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
             }
         } while (selected && exist && !overwrite);
         if (selected) {
+            const temp = new URI(uri.toString() + '(copy).tmp');
```
Comment (vince-fugnitto) [inline packages/workspace/src/browser/workspace-frontend-contribution.ts:420]: The variable `temp` is not used outside of the `try` block, therefore I think it should be placed inside.

File: packages/workspace/src/browser/workspace-frontend-contribution.ts [hunk 2]
```diff
@@ -417,12 +417,18 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
...
                 await this.commandRegistry.executeCommand(CommonCommands.SAVE.id);
-                await this.fileService.copy(uri, selected, { overwrite });
+                await this.fileService.move(uri, selected, { overwrite });
+                overwrite = true;
+                await this.fileService.move(temp, uri, { overwrite });
             } catch (e) {
                 console.warn(e);
             }
+            await this.commandRegistry.executeCommand(CommonCommands.CLOSE_TAB.id);
+            await open(this.openerService, selected);
```
Comment (vince-fugnitto) [inline packages/workspace/src/browser/workspace-frontend-contribution.ts:431]: Do we want to perform these operations even if there is an error in the `try-catch`?
Would it not be more valid to only perform these operations if the `save` was successful?

File: packages/workspace/src/browser/workspace-frontend-contribution.ts (hunk 1 above)
Comment (leopoldwe) [reply packages/workspace/src/browser/workspace-frontend-contribution.ts]: Ok. I moved it inside.

File: packages/workspace/src/browser/workspace-frontend-contribution.ts (hunk 2 above)
Comment (leopoldwe) [reply packages/workspace/src/browser/workspace-frontend-contribution.ts]: Yes, it makes more sense that way. I put the statements inside the try block.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 ## v1.6.0
 
 - [core] added functionality for handling context menu for `tab-bars` without activating the shell tab-bar [#6965](https://github.com/eclipse-theia/theia/pull/6965)
+- [workspace] "save as" command does not save changes to old file and opens the new file. [#8514](https://github.com/eclipse-theia/theia/pull/8514)
```
Comment (vince-fugnitto) [inline CHANGELOG.md:6]: Please be sure to update the `changelog` by rebasing it and placing it under the correct release (`1.7.0`).

**Review by vince-fugnitto — CHANGES_REQUESTED** [review]
The changes do no work well unfortunately, once a `save as` is performed the `explorer` is closed.

1. perform a `save as` choosing a new location under the same workspace.
2. the explorer is randomly closed after the operation.

**Review by vince-fugnitto — COMMENTED** [review]
@leopoldwe the current approach does not work well, and results in weird behaviors:
- perform a `save as` on a file and choose a location in the same workspace
- the `(copy)` file is visible in the UI (I would not expect this as an end-user)
- the original file is also marked as `Deleted` (by source control)

save-as

**Comment by leopoldwe** [comment]
I've tried to fix the source control problem, but I find it is outside of my abilities. I can't find a way to make the save as work and not a delete show in the source control temporarily.

**Pull request closed without merging**